lag/lag.yaml

```diff
@@ -29,12 +41,26 @@ components:
         ethernet:
           $ref: '../device/ethernet.yaml#/components/schemas/Device.EthernetBase'
 
+    Lag.Ports.Protocol:
+      type: object
+      properties:
+        choice:
+          description: >-
+            The type of controlling protocol for the LAG (ports group).
+          type: string
+          enum: [lacp, static]
+          default: lacp
+        lacp:
```

Comment:
Why is lacp being duplicated between ports and protocols?

Reply:
In OpenConfig or DUT, few LACP configuration properties are per LAG, not per port. In the existing model, such per LAG configuration is not allowed. Instead to achieve the same goal, the same set of configurations needs to be duplicated for each LAG member port. These could lead to inconsistency and undefined behavior. In IxNetwork, the modeling was done on per-port basis, as there was no option earlier to configure anything at Port Group level. Now as we are modeling in OTG, we can continue with the legacy model or we have the option to correct the model before it goes to the customer, the way it is used by the customer for other devices. So, few properties are duplicated to “ports” from “port” and properties at “port” level are marked deprecated.

Reply:
If LACP configuration is done at the LAG level does this mean configuration will never be done at the port level? Or is this just a convenience option?

Reply:
// Existing way of configuring LAG & LAG-member ports
lag1 := gosnappi.NewLag().SetName("lag1")
lag1.Ports().Add().SetPortName("lag1-p1").Protocol().Lacp().
		SetActorSystemId("01:01:01:01:01:01").SetActorSystemPriority(1).SetLacpduTimeout(1).SetActorPortNumber(1)
lag1.Ports().Add().SetPortName("lag1-p2").Protocol().Lacp().
		SetActorSystemId("01:01:01:01:01:01").SetActorSystemPriority(1).SetLacpduTimeout(1).SetActorPortNumber(2)
lag1.Ports().Add().SetPortName("lag1-p3").Protocol().Lacp().
		SetActorSystemId("01:01:01:01:01:01").SetActorSystemPriority(1).SetLacpduTimeout(1).SetActorPortNumber(3)

// New Proposal
lag1 = gosnappi.NewLag().SetName("lag1").Protocol().Lacp().SetActorSystemId("01:01:01:01:01:01").SetActorSystemPriority(1).SetLacpduTimeout(1)
lag1.Ports().Add().SetPortName("lag1-p1").Protocol().Lacp().SetActorPortNumber(1)
lag1.Ports().Add().SetPortName("lag1-p2").Protocol().Lacp().SetActorPortNumber(2)
lag1.Ports().Add().SetPortName("lag1-p3").Protocol().Lacp().SetActorPortNumber(3)

To avoid breaking old scripts, existing ways of configuring LAG-member ports are allowed (ideally we should make these lag-member attributes obsolete, instead of deprecate, but these are used by scripts for IxNetwork), but the recommendation is to configure LAG specific LACP parameters (i.e. ActorSystemId, ActorSystemPriority, LacpduTimeout & ActorActivity) at the LAG level, instead of repeating the same set of values for each LAG-member. And each LAG-member can not work with a different set of values, instead, any discrepancy of these parameters across LAG members are actually an invalid configuration.

Reply:
I see the difference now.  thanks.

Reply:
After discussing with Andy, per port protocol attributes that are now available per LAG, have been removed from per port level. Now we have just one way of configuration. The following properties are removed from per LAG-member (i.e. port):
- protocol choice
- lacp:
       actor_system_id, actor_system_priority, lacpdu_timeout, actor_activity
- static:
       lag_id

This is a breaking change.